src/qt/listctrl.cpp

```diff
@@ -980,9 +1039,146 @@ class wxQtListTreeWidget : public wxQtEventSignalHandler< QTreeView, wxListCtrl
+        {
+            setSectionsClickable(true);
+            setContextMenuPolicy(Qt::CustomContextMenu);
+            setSortIndicatorShown(true);
+
+            connect(this, &QHeaderView::sectionClicked,
+                    this, &wxQtHeaderView::sectionClicked);
+            connect(this, &QHeaderView::customContextMenuRequested,
+                    this, &wxQtHeaderView::sectionRightClicked);
+            connect(this, &QHeaderView::sectionResized,
+                    this, &wxQtHeaderView::sectionResized);
+        }
+
+    protected:
+        virtual void mouseReleaseEvent(QMouseEvent* event) override
+        {
+            if ( m_isDragging )
+            {
+                m_isDragging = false;
+
+                const ListEventData data =
+                    std::make_pair(m_parent->columnAt(event->x()), -1);
+
+                m_parent->EmitListEvent(wxEVT_LIST_COL_END_DRAG, QModelIndex(), &data);
+            }
+
+            QHeaderView::mouseReleaseEvent(event);
+        }
+
+    private:
+        void sectionClicked(int logicalIndex)
+        {
+            const ListEventData data = std::make_pair(logicalIndex, -1);
+
+            m_parent->EmitListEvent(wxEVT_LIST_COL_CLICK, QModelIndex(), &data);
+        }
+
+        void sectionRightClicked(const QPoint& pos)
+        {
+            const ListEventData data =
+                std::make_pair(m_parent->columnAt(pos.x()), -1);
+
+            m_parent->EmitListEvent(wxEVT_LIST_COL_RIGHT_CLICK, QModelIndex(), &data);
+        }
+
+        void sectionResized(int logicalIndex, int oldSize, int newSize)
+        {
+            ListEventData data;
+
+            if ( m_isDragging )
+            {
+                data = std::make_pair(logicalIndex, newSize);
+                m_parent->EmitListEvent(wxEVT_LIST_COL_DRAGGING, QModelIndex(), &data);
+                return;
+            }
+
+            if ( !underMouse() )
+            {
+                // sectionResized() also called if the control is being resized and
+                // the last section in the header is strechable, so just return and
+                // do nothing in this case.
+                return;
+            }
+
+            data = std::make_pair(logicalIndex, oldSize);
+
+            if ( sectionResizeMode(logicalIndex) == QHeaderView::Fixed ||
+                 !m_parent->EmitListEvent(wxEVT_LIST_COL_BEGIN_DRAG, QModelIndex(), &data) )
+            {
+                wxQtEnsureSignalsBlocked blocker(this);
+                resizeSection(logicalIndex, oldSize);
+                // This only takes effect after wxEVT_LIST_COL_END_DRAG is generated,
+                // after which the user can no longer drag the column.
+                setSectionResizeMode(logicalIndex, QHeaderView::Fixed);
+                return;
+            }
+
+            m_isDragging = true;
+        }
+
+        wxQtListTreeWidget* const m_parent;
+
+        bool m_isDragging;
```

Comment:
I'd initialize it here because there is just no reason not to:
```suggestion
        bool m_isDragging = false;
```
(and then we could remove its initialization from ctor member initializer list, of course).